schemas/post-artifact-request.yml

```diff
@@ -31,11 +31,35 @@ oneOf:
            always using `application/json` for JSON artifacts.
         type:       string
         maxLength:  255
+      contentMD5:
+        description: |
+          Base64 encoded MD5 sum of the artifact. If specified, it **must**
+          also be given as value to the `Content-MD5` header when uploading
+          to S3.
+          **Warning**, this will likely be required in the future, and **must**
+          be specified if `contentSHA256` is specified.
+        type:       string
```

Comment:
No worries, this was my misunderstanding, I thought the hex string was being base64 encoded, I didn't realise the binary hash value was being base64 encoded. Since shas and md5s are traditionally represented in hex, if it wasn't for s3 expecting it in base64, i would have found it more conventional to represent in hex. That said, could we convert between base64 and hex, so that client-interfacing values are in hex for both md5 and sha256, even if we pass through to s3 in base64? This just seems more canonical. Unless clients handle the signed urls and need to set s3 headers themselves, no point forcing them to use both.

Probably if one is base64 encoded, we should no doubt base64 encode both for consistency.